client/configuration_set_test.go

```diff
@@ -0,0 +1,200 @@
+			Expect(*configurationSet).To(Equal(mockConfigurationSet))
+		})
+	})
+
+	Describe("update configuration set", func() {
+		BeforeEach(func() {
+			updatePayload := UpdateConfigurationSetPayload{
+				Name:        "name2",
+				Description: "des2",
+			}
+
+			httpCall = mockHttpClient.EXPECT().
+				Put("/configuration-sets/"+id, &updatePayload, gomock.Any()).
+				Do(func(path string, request interface{}, response *ConfigurationSet) {
+					*response = mockConfigurationSet
+				}).Times(1)
+
+			configurationSet, _ = apiClient.ConfigurationSetUpdate(id, &updatePayload)
+		})
+
+		It("Should return configuration set", func() {
+			Expect(*configurationSet).To(Equal(mockConfigurationSet))
+		})
+	})
+
+	Describe("list organization scope configuration sets", func() {
+		BeforeEach(func() {
+			mockOrganizationIdCall(organizationId).Times(1)
+
+			httpCall = mockHttpClient.EXPECT().
+				Get("/configuration-sets", map[string]string{
+					"scope":               "organization",
+					"scopeId":             organizationId,
+					"includeHigherScopes": "false",
+				}, gomock.Any()).
+				Do(func(path string, request interface{}, response *[]ConfigurationSet) {
+					*response = mockConfigurationSetList
+				}).Times(1)
+
+			configurationSetList, _ = apiClient.ConfigurationSets("organization", "")
+		})
+
+		It("Should return configuration set list", func() {
+			Expect(configurationSetList).To(Equal(mockConfigurationSetList))
+		})
+	})
+
+	Describe("list project scope configuration sets", func() {
+		BeforeEach(func() {
+			httpCall = mockHttpClient.EXPECT().
+				Get("/configuration-sets", map[string]string{
+					"scope":               "project",
+					"scopeId":             projectId,
+					"includeHigherScopes": "false",
+				}, gomock.Any()).
+				Do(func(path string, request interface{}, response *[]ConfigurationSet) {
+					*response = mockConfigurationSetList
+				}).Times(1)
+
+			configurationSetList, _ = apiClient.ConfigurationSets("project", projectId)
+		})
+
+		It("Should return configuration set list", func() {
+			Expect(configurationSetList).To(Equal(mockConfigurationSetList))
+		})
+	})
+
+	Describe("get configuration set by id", func() {
+		BeforeEach(func() {
+			httpCall = mockHttpClient.EXPECT().
+				Get("/configuration-sets/"+id, nil, gomock.Any()).
+				Do(func(path string, request interface{}, response *ConfigurationSet) {
+					*response = mockConfigurationSet
+				}).Times(1)
+
+			configurationSet, _ = apiClient.ConfigurationSet(id)
+		})
+
+		It("Should return configuration set", func() {
+			Expect(*configurationSet).To(Equal(mockConfigurationSet))
+		})
+	})
+
+	Describe("delete configuration set", func() {
+		BeforeEach(func() {
+			httpCall = mockHttpClient.EXPECT().
+				Delete("/configuration-sets/"+id, nil).
+				Do(func(path string, request interface{}) {}).
+				Times(1)
+
+			apiClient.ConfigurationSetDelete(id)
+		})
+
+		It("Should call delete once", func() {})
```

Comment:
its empty...

Reply:
don't you prefer to do in in the test not in the before each?

Reply:
this will test that delete was called once.
Without it, it test's nothing...